apis/postgresql/v1alpha1/provider_types.go

```diff
@@ -26,6 +26,10 @@ import (
 type ProviderConfigSpec struct {
 	// Credentials required to authenticate to this provider.
 	Credentials ProviderCredentials `json:"credentials"`
+	// Defines the database name used to set up a connection to the provided
+	// PostgreSQL instance. Same as PGDATABASE environment variable.
+	// +kubebuilder:default="postgres"
+	DBName string `json:"dbName"`
```

Comment:
Is there a more descriptive name we could use for this field? It does look like PostgreSQL calls this the `dbname`, but is it fair to say that we always want to connect to the "default database" or something similar? It seems a little strange to have to pass in a "database name" when the operation we want to undertake might well be creating a new database. 🤔

Reply:
Good point. What about `connectionDatabase` given it is used in the connection (dsn) string? 
But it could also be overridden by the spec of `Extension` type, so `defaultDatabase` could be better considering that.